chainstate/tx-verifier/src/transaction_verifier/error.rs

```diff
@@ -145,6 +145,66 @@ pub enum ConnectTransactionError {
     OutputTimelockError(#[from] timelock_check::OutputMaturityError),
 }
 
+impl ConnectTransactionError {
+    pub fn is_utxo_missing_error(&self) -> bool {
```

Comment:
I would prefer this is moved to the orphan transaction part of the code because it's irrelevant here in the general context. We discourage generalizing such ideas because it can lead to negligence errors. A good example is a "value" of an output, which isn't always clear depending on the output type.

Reply:
I'm sorry, but this can encourage abusing it just due to its name. We have to be extremely careful with what other devs can do wrong. Please move this to the mempool if this is ought to be used in the mempool.

And just to be clear: It's not like these kinds of mistakes never happen. We had enough instances of people using the `value()` function and `destination()` methods of the output struct/enum incorrectly until we decided that they have to go, to force people to match on the enum, because most devs don't even understand the context and assume the simplest explanation possible from the name of the function. Then tons of clean up had to be done to fix all the context mistakes due to that.